todo_app/tests_e2e/test_selenium.py

```diff
@@ -0,0 +1,360 @@
+import os
+from dotenv import load_dotenv, find_dotenv
+from threading import Thread
+import requests
+import pytest
+from todo_app.trello import Trello
+from todo_app import app
+from selenium import webdriver 
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions 
+from selenium.webdriver.common.by import By
+from selenium.webdriver import ActionChains
+
+@pytest.fixture()
+def driver():
+    with webdriver.Firefox() as driver:
+        yield driver
+
+@pytest.fixture()
+def test_app():
+    
+    file_path = find_dotenv('.env.test.e2e')
+    load_dotenv(file_path, override=True)
+
+    os.environ['API_KEY'] = os.getenv('API_KEY')
+    os.environ['API_TOKEN'] = os.getenv('API_TOKEN')
+    
+    
+    board_id = create_trello_board("TestBoard")
+    os.environ['TRELLO_BOARD_ID'] = board_id
+    os.environ['BOARD_NAME'] = "TestBoard"
+    os.environ['TO_DO_LIST_ID'] = get_list_id_for_board(board_id, "TO DO")
+    os.environ['DONE_LIST_ID'] = get_list_id_for_board(board_id, "DONE")
+    os.environ['DOING_LIST_ID'] = get_list_id_for_board(board_id, "DOING")
+    
+    # construct the new application
+    application = app.create_app()
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False))
+    thread.daemon = True
+    thread.start()
+    yield app
+    # Tear Down
+    thread.join(1)
+    delete_trello_board(board_id)
+
+def create_trello_board(board_name):
+    trello = Trello()
+    create_board_url = trello.add_key_and_token("https://api.trello.com/1/boards?name=" + board_name + "&")
+    request = requests.post(create_board_url)
+    response = request.json()
```

Comment:
Your `request` variable is actually an http response. Your `response` variable is the parsed body of the response